core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -176,19 +176,34 @@ public void initialize(String name, Map<String, String> unresolved) {
     long startTimeMillis =
         System.currentTimeMillis(); // keep track of the init start time for token refresh
     String initToken = props.get(OAuth2Properties.TOKEN);
+    boolean hasInitToken = initToken != null;
 
     // fetch auth and config to complete initialization
     ConfigResponse config;
     OAuthTokenResponse authResponse;
     String credential = props.get(OAuth2Properties.CREDENTIAL);
+    boolean hasCredential = credential != null && !credential.isEmpty();
     String scope = props.getOrDefault(OAuth2Properties.SCOPE, OAuth2Properties.CATALOG_SCOPE);
     Map<String, String> optionalOAuthParams = OAuth2Util.buildOptionalParam(props);
+    if (!props.containsKey(OAuth2Properties.OAUTH2_SERVER_URI)
```

Comment:
`initialize` is already quite long and the if condition is complicated here, so might be worth extracting this into a separate method

Reply:
Is this _really_ important or rather a nit for code that is going to be changes in the near future?

Reply:
yeah I think if we want to formally refactor this, it might be okay to leave the warning here for now. What do you think Eduard?

Reply:
I'm ok leaving it as-is but it adds complexity to the already difficult to understand `initialize` method, hence why I suggested to consider moving this into a separate method (even if the code eventually gets changed in the near future we don't necessarily have to add additional complexity if we can avoid it in the first place)

Reply:
I am doing a prototype for the refactoring, maybe let me publish a draft and we can see how that would look like

Reply:
I'd rather let this change not depend on a follow-up, but get it in.